### ISSUE TITLE
Liquid freshness of the whole barrel is defined by the first liquid poured instead of an average

### ISSUE THREAD

**Issue by Guimoute** [anegostudios/VintageStory-Issues#2839]
### Game Version

v1.18.5

### Platform

Windows

### Modded

Modded

### SP/MP

Singleplayer

### Description

(None of my mods affect barrels)

As shown in the video, if you pour any amount of fresh liquid in a barrel, the spoiled liquid subsequently added is considered fresh and adopts the remaining shelf life of the initial liquid. Althought untested, I assume it works the other way around too.  

https://github.com/anegostudios/VintageStory-Issues/assets/51537797/60fb1718-c6b0-4cbf-a4c4-31980747a86a

### Logs

```Log```

**Comment by Guimoute** [comment]
Funnily enough, if you stagger the pregnancies of your ewes, this bug means you can keep your entire stock of milk fresh forever if you don't mind moving the content of your barrels.

**Comment by Guimoute** [comment]
Still in 1.19.8.

**Comment by Guimoute** [comment]
Confirmed in 1.20.3.

**Comment by pingouyn** [comment]
Yes, and I've used it as an exploit for a year...  I confess, sorry.  Would be nice if it's fixed though, as probably not an intended behaviour.

**Comment by Craluminum2413** [comment]
@daveboy2000 What about 1.21.0-rc.6?

**Comment by sekelsta** [comment]
Probably related to https://github.com/anegostudios/VintageStory-Issues/issues/1569

**Comment by pizza2004** [comment]
Fixed in 1.22.
